internal/services/rate.go

```diff
@@ -68,29 +76,37 @@ func (rs *CachingRateService) StartBackgroundTask() {
 	}()
 }
 
-func (rs *CachingRateService) GetRate() (*Rate, error) {
+func (rs *CachingRateService) GetRate() (float32, error) {
 	rs.mutex.RLock()
 	defer rs.mutex.RUnlock()
 
 	if rs.fetchError != nil {
-		return nil, rs.fetchError
+		return 0, rs.fetchError
 	}
 	return rs.currentRate, nil
 }
 
-func (rs *CachingRateService) fetchExchangeRate() (*Rate, error) {
-	resp, err := http.Get(RatesAPIBaseURL + "/usd.json")
+type HTTPExchangeRateClient struct {
+	client *http.Client
+}
+
+func NewHTTPExchangeRateClient(client *http.Client) *HTTPExchangeRateClient {
+	return &HTTPExchangeRateClient{client: client}
+}
+
+func (ec *HTTPExchangeRateClient) GetExchangeRate() (float32, error) {
```

Comment:
Do you mind making client flexible to be able return exchange rate of different currency pairs?

Reply:
Yes, I'm going to extend rate clients capabilities in the next homework, as well as creating several additional exchange rate providers.